Jellyfin.Data/Entities/User.cs

```diff
@@ -413,6 +413,52 @@ public string[] GetPreference(PreferenceKind preference)
             return Equals(val, string.Empty) ? Array.Empty<string>() : val.Split(Delimiter);
         }
 
+        /// <summary>
+        /// Gets the user's preferences for the given preference kind.
+        /// </summary>
+        /// <param name="preference">The preference kind.</param>
+        /// <typeparam name="T">Type of preference.</typeparam>
+        /// <returns>A {T} array containing the user's preference.</returns>
+        public T[] GetPreferenceValues<T>(PreferenceKind preference)
+        {
+            var val = Preferences.First(p => p.Kind == preference).Value;
+            if (string.IsNullOrEmpty(val))
+            {
+                return Array.Empty<T>();
+            }
+
+            // Convert array of {string} to array of {T}
+            var converter = TypeDescriptor.GetConverter(typeof(T));
+            var stringValues = val.Split(Delimiter);
+            var parsedValues = new object[stringValues.Length];
+            var convertedCount = 0;
+            for (var i = 0; i < stringValues.Length; i++)
```

Comment:
Couldn't this just be rewritten as this? Not sure what the range indices do under the hood regarding allocations etc. though
```C#
            for (var i = 0; i < stringValues.Length; i++)
            {
                try
                {
                    var parsedValue = converter.ConvertFromString(stringValues[i].Trim());
                    if (parsedValue != null)
                    {
                        parsedValues.SetValue(parsedValue, convertedCount++);
                    }
                }
                catch (FormatException)
                {
                    // Unable to convert value
                }
            }

            return parsedValues[..convertedCount];
```

Reply:
Internal impl : https://github.com/dotnet/runtime/blob/804d730d2fd8a57dbe9c0a3118800b1288993ecd/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs#L23